protobuf-kafkaconnect-converter/src/main/java/com/amazonaws/services/schemaregistry/kafkaconnect/protobuf/toconnectdata/ProtobufDataToConnectDataConverter.java

```diff
@@ -0,0 +1,107 @@
+package com.amazonaws.services.schemaregistry.kafkaconnect.protobuf.toconnectdata;
+
+import com.google.protobuf.ByteString;
+import com.google.protobuf.Descriptors;
+import com.google.protobuf.Message;
+import lombok.NonNull;
+import org.apache.kafka.connect.data.Field;
+import org.apache.kafka.connect.data.Schema;
+import org.apache.kafka.connect.data.Struct;
+import org.apache.kafka.connect.errors.DataException;
+
+import java.util.List;
+
+/**
+ * Converts Protobuf data to Connect data corresponding to the translated schema.
+ */
+public class ProtobufDataToConnectDataConverter {
+    public Object toConnectData(@NonNull final Message message, @NonNull final Schema connectSchema) {
+
+        final List<Field> fields = connectSchema.fields();
+        final Struct data = new Struct(connectSchema);
+        fields.forEach(
+            (field) -> toConnectDataField(field, message, data));
+
+        return data;
+    }
+
+    private void toConnectDataField(
+        final Field connectField, final Message message, final Struct data) {
+        final Schema connectSchema = connectField.schema();
+        final Descriptors.FieldDescriptor fieldDescriptor = getFieldByName(message, connectField.name());
+
+        if (fieldDescriptor == null) {
+            throw new DataException("Protobuf schema doesn't contain the connect field: " + connectField.name());
+        }
+
+        final Object value = message.getField(fieldDescriptor);
+        //Unfortunately Protobuf 3 has a complex way to check for optionals.
+        final boolean isOptionalFieldNotSet =
```

Comment:
How do we handle proto3 optional here?

Reply:
The method `hasOptionalKeyword` takes care of it, [Reference](https://github.com/protocolbuffers/protobuf/blob/master/java/core/src/main/java/com/google/protobuf/Descriptors.java#L1261)